content/articles/en/challenges-with-switching-to-pinia.md

```diff
@@ -0,0 +1,288 @@
+export const useProfileStore = defineStore('profile', {
+  state: (): State => ({
+    user: {},
+    options: {},
+  }),
+
+  actions: {
+    // mutations can now become actions. Instead of `state` as the first argument, use `this`
+    setUser(user) {
+      this.user = user
+    },
+
+    setOptions(payload) {
+      const { key, value } = payload
+      this.options[key] = value
+    },
+
+    // Now we can get rid of the rootGetters and getters functions!
+    async retrievePendingUserSurveys(
+       { force = false } = {}
+     ) {
+       const userSurvey = surveyStore.surveys.filter(s => s.userId === this.user.id)
+       return userSurvey
+    }
+  },
+})
+```
+
+If anything, I do like how Pinia is used to create a store. Also, `Pinia` uses `stores` instead of `store` as part of its folder structure. You can see more about that here in their migration guide: https://pinia.vuejs.org/cookbook/migration-vuex.html#restructuring-modules-to-stores
+
+Now, this makes sense. If we are going to be naming our stores with the `use` prefix, then it makes sense that the folder is named `stores`.
+
+Using Pinia's store API in practice makes things a little more difficult ( or perhaps just different ) when you compare it to Vuex.
+
+## How we utilize Vuex in code
+
+If we want to use our profile store in a page, we do not have to import anything. This is because the state is accessed with the `$store` property. This can be accessed anywhere in our site using `this.$store`.
+
+The exception is when we want to use it in middleware, `asyncData`, or other lifecycle hooks that do not have access to the store. In these cases, we normally pass the `store` property. For example:
+
+```js
+asyncData(axios, store) {
+  const currentProfile = store.profile.user
+  ...
+}
+```
+
+## How does this change in Pinia?
+
+Pinia makes things just a little more difficult and a little more verbose.
+
+Let's take the most annoying part. If we want to access the store in `asyncData`, we cannot just pass `store` ( in fact, it would not longer exist with pinia ). Instead, we have to pass `$pinia`.
+
+```js
+asyncData(axios, $pinia) {
+  ...
+}
+```
+
+But that's not all. We _also_ have to import the store that we want to use, and then pass the `$pinia` property to the store function to "activate" it.
+
+```js
+import useProfileStore from '~stores/profile'
+
+...
+
+asyncData(axios, $pinia) {
+  const profileStore = useProfileStore($pinia)
+
+  // now we can access our states
+  const user = profileStore.user
+  const options = profileStore.options
+}
+```
+
+Whether you think this is better or worse depends on your idea of ease of use and maintainability. Personally, I think it makes things a little harder to use since we now have to import all the stores that we need to access in a particular file. Basically, it takes longer to set up. However, this may mean that maintainability may be a little better ( not necessarily easier ), but that depends on you.
```

Comment:
[Typography]

No space before `)`
No sapce after `(`.

This looks weird.

![image](https://user-images.githubusercontent.com/14120117/164384210-ffbf6263-fbb6-464e-8cc1-40ab23c88339.png)

Reply:
Fixed.